packages/itwin/quantity-formatting/src/components/quantityformat/FormatSelector.tsx

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as React from "react";
+
+import type { FormatDefinition } from "@itwin/core-quantity";
+import { ComboBox, Flex, Label, Text } from "@itwin/itwinui-react";
+
+import { useTranslation } from "../../useTranslation.js";
+
+import type { FormatSet } from "@itwin/ecschema-metadata";
+
+interface FormatSelectorProps {
+  activeFormatSet?: FormatSet;
+  activeFormatDefinitionKey?: string;
+  onListItemChange: (formatDefinition: FormatDefinition, key: string) => void;
+}
+
+export const FormatSelector: React.FC<FormatSelectorProps> = ({
+  activeFormatSet,
+  activeFormatDefinitionKey,
+  onListItemChange,
+}) => {
+  const { translate } = useTranslation();
+  const formatListId = React.useId();
+
+  // Prepare options for ComboBox
+  const formatOptions = React.useMemo(() => {
+    if (!activeFormatSet?.formats) {
+      return [];
+    }
+
+    return Object.entries(activeFormatSet.formats).map(([key, formatDef]) => ({
+      value: key,
+      label: formatDef.label || key,
+      sublabel: formatDef.description,
+    }));
+  }, [activeFormatSet?.formats]);
+
+  const handleFormatChange = React.useCallback(
+    (value: string) => {
+      if (activeFormatSet?.formats && value) {
+        const formatDef = activeFormatSet.formats[value];
+        if (formatDef) {
+          onListItemChange(formatDef, value);
+        }
+      }
+    },
+    [activeFormatSet?.formats, onListItemChange],
+  );
+
+  return (
+    <Flex flexDirection="column" alignItems="flex-start">
+      {activeFormatSet && (
+        <>
+          <Label htmlFor={formatListId} className="quantity-format-label">
+            <Text variant="leading">
+              {translate("QuantityFormat:labels.formatSetSectionLabel")}
+            </Text>
+          </Label>
+          <ComboBox
+            id={formatListId}
+            className="format-selector-list"
+            options={formatOptions}
+            value={activeFormatDefinitionKey || ""}
+            onChange={handleFormatChange}
+            enableVirtualization
+            dropdownMenuProps={{
+              style: {
+                maxBlockSize: "min(calc((var(--iui-component-height) - 1px) * 30), 70vh)",
```

Comment:
this is bit of a doozy to be hardcoding

Reply:
The [default](https://github.com/iTwin/iTwinUI/blob/bbf478f2bcae8fca3e483020e0e048aee37f4dde/packages/itwinui-css/src/menu/menu.scss#L26) is configured to only be around 8 items, and based on the number of KoQs any normal imodel has, 8 is not enough, so I increased it to 30 items. If the dropdown is larger then we restrict the height to be 70% of the screen's height. I can leave a comment explaining why this was done

Reply:
thanks, your recommendation got the className approach to work. I reduced it to 20.5, and got rid of 70 vh